clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -872,6 +875,10 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void unsubscribe() {
         acquire();
         try {
+            // make sure the offsets of topic partitions the consumer is unsubscribing from
+            // are committed since there will be no following rebalance
+            this.coordinator.maybeAutoCommitOffsetsAsync(Long.MAX_VALUE);
```

Comment:
Sorry, I made a comment on this yesterday but became hidden after I submitted an update. Here's the comment:

> I am running into an issue with your suggestion in the second bullet. The method maybeAutoCommitOffsetsAsync(...) takes a long, and when I pass in time.milliseconds() (i.e. calling this.coordinator.maybeAutoCommitOffsetsAsync(time.milliseconds()); here) it doesn't execute the main block of code because the condition now >= nextAutoCommitDeadline does not hold. It seems we need to pass in a long value that's greater than the next auto-commit time (in milliseconds). Since we just want to make sure the main block of code in that method (which involves committing the offsets) is executed, we could just pass Long.MAX_VALUE; though I'm not sure if that's the cleanest approach.

If we create a public `maybeAutoCommitOffsetAsync` that calls the old private method we would still need to pass in the `long` value. Unless the public one repeats the code inside the private method without the timestamp handling?